src/promptflow/tests/sdk_cli_test/unittests/test_exceptions.py

```diff
@@ -80,12 +108,77 @@ def test_error_category_with_executor_error(self):
+
+        ex = None
+        try:
+            try:
+                FlowValidator._validate_aggregation_inputs({}, {"input1": "value1"})
+            except Exception as e:
+                raise UserErrorException("FlowValidator._validate_aggregation_inputs failed") from e
+        except Exception as e:
+            ex = e
+        error_category, error_type, error_target, error_message = _ErrorInfo.get_error_info(ex)
+        assert error_category == ErrorCategory.SystemError
+        assert error_type == "InvalidAggregationInput"
+        assert error_target == ErrorTarget.UNKNOWN
+        assert (
+            "exception msg=The input for aggregation is incorrect. "
+            "The value for aggregated reference input '{input_key}' "
+            "should be a list, but received {value_type}. "
+            "Please adjust the input value to match the expected format., "
+            "exception module=promptflow.executor.flow_validator, "
+            "exception code=raise InvalidAggregationInput(, "
+            "exception lineno="
+        ) in error_message
+
+    def test_error_category_with_cause_exception2(self, pf):
+        """cause exception is not PromptflowException and e is PromptflowException, recording e exception."""
+
+        ex = None
+        try:
+            try:
+                pf.run("./exceptions/flows")
+            except Exception as e:
+                raise UserErrorException("pf run failed") from e
+        except Exception as e:
+            ex = e
+        error_category, error_type, error_target, error_message = _ErrorInfo.get_error_info(ex)
+        assert error_category == ErrorCategory.UserError
+        assert error_type == "UserErrorException"
+        assert error_target == ErrorTarget.UNKNOWN
+        assert (
+            "exception msg=Non promptflow message, not recorded., "
+            "exception module=Non promptflow module, not recorded., "
+            "exception code=Non promptflow code, not recorded., "
+            "exception lineno=Non promptflow code lineno, not recorded."
+        ) in error_message
+
+    def test_error_category_with_cause_exception3(self, pf):
+        """cause exception is not PromptflowException and e is not PromptflowException, recording cause exception."""
+
+        ex = None
+        try:
+            try:
+                pf.run("./exceptions/flows")
+            except Exception as e:
+                raise Exception("pf run failed") from e
+        except Exception as e:
+            ex = e
+        error_category, error_type, error_target, error_message = _ErrorInfo.get_error_info(ex)
+        assert error_category == ErrorCategory.UserError
+        assert error_type == "FileNotFoundError"
+        assert error_target == ErrorTarget.UNKNOWN
+        assert (
+            "exception msg=, "
+            "exception module=promptflow._sdk._pf_client, "
+            'exception code=raise FileNotFoundError(f"flow path {flow} does not exist"), '
```

Comment:
looks like this dulp with msg, we can remove code

Reply:
According to our latest discussion, I have split it into two fields.